.github/workflows/smoke-test.yml

```diff
@@ -0,0 +1,87 @@
+name: Smoke Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - main
+  merge_group:
```

Comment:
Should we still run this on pull request & push? Isn't merge group enough?

Reply:
Yes, it will be sufficient. I did mark this as ready for review too early, I suppose, since I'm still thinking of other solutions. ;p I wanted to cut the time somebody has to wait before getting feedback on whether the environments are good or not by running it as part of the PR.

Reply:
Yes, it will be sufficient. I did mark this as ready for review too early, I suppose, since I'm still thinking of other solutions. ;p I wanted to cut the time somebody has to wait before getting feedback on whether the environments are good or not by running it as part of the PR.